pkg/transaction/snapper_update_helper.go

```diff
@@ -0,0 +1,302 @@
+/*
+Copyright © 2022-2025 SUSE LLC
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transaction
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/suse/elemental/v3/pkg/btrfs"
+	"github.com/suse/elemental/v3/pkg/chroot"
+	"github.com/suse/elemental/v3/pkg/deployment"
+	"github.com/suse/elemental/v3/pkg/fstab"
+	"github.com/suse/elemental/v3/pkg/rsync"
+	"github.com/suse/elemental/v3/pkg/snapper"
+	"github.com/suse/elemental/v3/pkg/sys/vfs"
+	"github.com/suse/elemental/v3/pkg/unpack"
+)
+
+func (sc snapperContext) SyncImageContent(imgSrc *deployment.ImageSource, trans *Transaction, fullSync bool) (err error) {
```

Comment:
The methods of this file are essentially the former `transaction.Update` method splitting each step in an independent method. This allows further granularity to execute additional logic between those and it removes the need of the upgrade callbacks.